lib/client.js

```diff
@@ -97,6 +97,10 @@ WorkerClient.prototype.registerChannel = function(channel) {
                 if (ch.output) {
                     assertedOutputs[ch.output] = qok.queue;
                 }
+                // do not consume job results when consumeJobResults === false
+                if (ch.output && !workerClient.commander.canConsumeJobResults()) {
```

Comment:
`ch.output` is the same as `name` afaics. Perhaps the `name` variable should be removed from this function and only call `assertQueueWithHandler(ch)`.